src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,30 +1,35 @@
 # TO-DO: Complete the selection_sort() function below 
-def selection_sort( arr ):
+def selection_sort( arr ):   
     # loop through n-1 elements
-    for i in range(0, len(arr) - 1):
-        cur_index = i
-        smallest_index = cur_index
+    for i in range(0, len(arr) - 1):     
+        cur_index = i                    
+        smallest_index = cur_index 
         # TO-DO: find next smallest element
         # (hint, can do in 3 loc) 
-             
-
-
-
-        # TO-DO: swap
-
-
-
+        for x in range(cur_index, len(arr)):
+            if arr[x] < arr[smallest_index]:
+                smallest_index = x
+       # TO-DO: swap
+        temp = arr[smallest_index]
+        
+        arr[smallest_index] = arr[cur_index]
+       
+        arr[cur_index] = temp
+       
 
     return arr
 
 
 # TO-DO:  implement the Bubble Sort function below
 def bubble_sort( arr ):
-
+    for num in range(len(arr)-1, 0, -1):
```

Comment:
Like he talked about in lecture, instead of going through a specific number of times, you can do a while loop and go through until you don't do any swaps in one pass :)